Python3.13 support

73VW opened the pull request:
This fixes the issue https://github.com/pajachiet/pymongo-schema/issues/36.

## Devcontainer

- Add compose-based devcontainer with `mongodb` service (mongo:7)
- Share network namespace for `localhost:27017` connectivity
- Post-create imports functional fixtures via `scripts/init_mongo_from_json.py`
- Document Python 3.13 + ete4 support and test workflow
- Note Extended JSON parsing with `bson.json_util`"

## Migration to ete 4 and dependencies modernization

- Replace ete3 with ete4 (`setup.py`, `requirements.txt`)
- Build type tree programmatically and use `common_ancestor()`
- Update pandas testing import (`pandas.testing.assert_frame_equal`)
- Simplify xlsx export to use `DataFrame.to_excel()` with openpyxl
- Remove legacy Python 2.7 `Dockerfile` (obsolete)

## Refactoring

- Make type tree construction explicit for ete4

## Packaging

- Align packaging to python3.13

73VW committed:
chore(devcontainer): add Dev Container with MongoDB sidecar and auto fixtures import

docs(readme): add Manual Testing and Contributing sections

- Add compose-based devcontainer with `mongodb` service (mongo:7)
- Share network namespace for `localhost:27017` connectivity
- Post-create imports functional fixtures via `scripts/init_mongo_from_json.py`
- Document Python 3.13 + ete4 support and test workflow
- Note Extended JSON parsing with `bson.json_util`"

73VW committed:
fix(compat): migrate to ete4 and modernize dependencies

- Replace ete3 with ete4 (`setup.py`, `requirements.txt`)
- Build type tree programmatically and use `common_ancestor()`
- Update pandas testing import (`pandas.testing.assert_frame_equal`)
- Simplify xlsx export to use `DataFrame.to_excel()` with openpyxl
- Remove legacy Python 2.7 `Dockerfile` (obsolete)

refactor(types): make type tree construction explicit for ete4
chore: align packaging to Python 3.13"

pajachiet commented:
Hey,

Thanks for the issue and the PR, and for all the work you’ve done — this looks great!

I’m no longer able to properly review, merge, and publish a new version.

@andrii0yerko: would you be able to take care of it?

andrii0yerko commented on README.md:
```diff
@@ -361,3 +361,45 @@ To tackle bigger databases, it certainly would be usefull to implement the follo
+- Development tips:
+    - Run `scripts/init_mongo_from_json.py` to load fixtures.
+    - Pandas testing uses `pandas.testing.assert_frame_equal`.
+    - ETE Tree API uses `common_ancestor()` and a programmatic type tree setup.
```
Looks like coding agent output to me

It's not very informative. Should we remove it?

73VW commented on scripts/init_mongo_from_json.py:
The main purpose was to do the same, without `mongoimport` command. I will change that.

73VW committed:
feat: Add mongodb-org-tools in devcontainer

- Remove useless python script
- Default to original db init script in test folder for container setup
- Remove unnecessary contributing section in README

73VW commented:
Hey @andrii0yerko!

Thank you for your review. I have made a few changes based on your feedback.

What do you think of this now?

andrii0yerko reviewed (approved):
Thanks! LGTM

andrii0yerko commented:
Hey @pajachiet 
I have no right to merge into the repo. Can you help with that, please? (merge or grant rights maybe)

Then I will publish the updated version on PyPi.

Thanks in advance!

pajachiet commented:
Thanks a lot !

Sorry I was not watching my emails. All the best.

sylvain1811 commented:
Hey @andrii0yerko, could you publish the updated package on PyPi by any chance? Thanks a lot!

andrii0yerko commented:
Done!

@sylvain1811 @73VW consider updating to 0.4.2 from pypi!
